enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj

```diff
@@ -62,6 +62,15 @@
         :present #{:card_id :attribute_remappings})))
   (db/select-one GroupTableAccessPolicy :id id))
 
+(api/defendpoint POST "/validate"
+  "Validate a sandbox which may not have yet been saved. This runs the same validation that is performed when the
+  sandbox is saved, but doesn't actually save the sandbox."
+  [:as {{:keys [table_id card_id]} :body}]
+  {table_id             pos-int?
+   card_id              [:maybe pos-int?]}
+  (gtap/check-columns-match-table {:table_id table_id
+                                   :card_id  card_id}))
```

Comment:
if `:card_id` is not passed into `check-columns-match-table`, then [no check will occur](https://github.com/metabase/metabase/blob/gtap-in-permissions-graph/enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj#L88-L96). 

Could you add a test for it, and is that intended behavior?

Reply:
`card_id` is optional, it is possible to create a policy without it

Reply:
So if it's optional, will any policy without one be valid? I'm not actually sure, but a test with a nil card_id would be nice to see.

Reply:
@escherize yeah this is intended. a sandboxing policy without a `card_id` is one that instead uses user attributes to filter on a column. and the validation we're currently doing is only relevant for sandboxing that uses an underlying card to replace a table.

I'll add a test that just checks that validation passes when `card_id` is nil.